server/bundler/loader-utils.js

```diff
@@ -0,0 +1,70 @@
+const fs = require( 'fs' ),
+	utils = require( './utils' );
+
+function getSectionsModule( sections, codeSplitting ) {
+	let templateName,
+		stringSections,
+		sectionLoaders;
+
+	if ( codeSplitting ) {
+		templateName = 'loader-code-splitting.js.template';
+		sectionLoaders = getSectionLoadersCodeSplitting( sections );
+	} else {
+		templateName = 'loader.js.template';
+		sectionLoaders = getSectionLoaders( sections );
+	}
+
+	stringSections = JSON.stringify( sections );
+
+	// Load the template and convert to a template literal using eval
+	return eval( '`' + fs.readFileSync( __dirname + '/' + templateName, 'utf8' ).toString() + '`' );
```

Comment:
I don't think `.template` files are making the code that much clearer imo. `JavaScript` being a script language we have other solutions for meta-programming. `( ${ myFunction.toString() } )( ${ myParams } );` being one of them for instance.

In the case of `loader-code-splitting.js`, you could easily add a module variable and transform `preload` into:

```
var sectionImports = { /* inserted dynamically */ };
var _loadedSections = {};

function preload( sectionName ) {
    if ( ! _loadedSections[ sectionName ] ) {
        _loadedSections[ sectionName ] = sectionImports[ sectionName ]();
    }
    
    return _loadedSections[ sectionName ];
}
```


```
var sectionImports = sections.map( section => {
	return `${ JSON.stringify( section.name ) }: function(){ return import( /* webpackChunkName: ${ JSON.stringify( section.name ) } */ ${ JSON.strinngify( section.module ) } ) }`;
}, {} );

return eval( fs.readFileSync( __dirname + '/' + templateName, 'utf8' ).toString().replace( 'var sectionImports = { /* inserted dynamically */ };', `var sectionImports = { ${ sectionImports.join(',\n') }; };` ) );
```

We could also probably get rid of `eval` if we generate this file on build time and require it later on.

Reply:
> We could also probably get rid of eval if we generate this file on build time and require it later on.

We should probably investigate this more. Having the section splitting logic closer to our webpack config sounds logical to me, it could probably be done with a webpack plugin or loader.